tokio/src/runtime/handle.rs

```diff
@@ -345,21 +352,21 @@ impl HandleInner {
         F: FnOnce() -> R + Send + 'static,
         R: Send + 'static,
     {
-        let (join_handle, spawn_result) = if cfg!(debug_assertions)
-            && std::mem::size_of::<F>() > 2048
-        {
+        let spawn_result = if cfg!(debug_assertions) && std::mem::size_of::<F>() > 2048 {
             self.spawn_blocking_inner(Box::new(func), blocking::Mandatory::NonMandatory, None, rt)
         } else {
             self.spawn_blocking_inner(func, blocking::Mandatory::NonMandatory, None, rt)
         };
 
         match spawn_result {
-            Ok(()) => join_handle,
-            // Compat: do not panic here, return the join_handle even though it will never resolve
-            Err(blocking::SpawnError::ShuttingDown) => join_handle,
-            Err(blocking::SpawnError::NoThreads(e)) => {
-                panic!("OS can't spawn worker thread: {}", e)
-            }
+            Ok(join_handle) => join_handle,
+            Err(e) => match e.inner.kind {
+                // Compat: do not panic here, return the join_handle even though it will never resolve
+                SpawnBlockingErrorKind::Shutdown => e.handle,
```

Comment:
That's not true. The join handle should immediately resolve to a cancelled error.

Reply:
Thanks will fix the comment